File: packages/node/src/http/httpClient.ts
import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';

export class HttpClient {
  private client: AxiosInstance;
  private baseUrl: string;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
    this.client = axios.create({
      baseURL: baseUrl,
    });
  }

  async post(url: string, body: any, token: string): Promise<any> {
    const config: AxiosRequestConfig = {
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    };

    try {
      const response = await this.client.post(url, body, config);
      return response.data;
    } catch (error: any) {
      throw new Error(`Mpesa request failed: ${error.response?.data?.errorMessage || error.message}`);
    }
  }

  async get(url: string, token: string): Promise<any> {
    const config: AxiosRequestConfig = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };

    try {
      const response = await this.client.get(url, config);
      return response.data;
    } catch (error: any) {
      throw new Error(`Mpesa request failed: ${error.response?.data?.errorMessage || error.message}`);
    }
  }

  getBaseUrl(): string {
    return this.baseUrl;
  }
}

File: packages/python/src/mpesa/http/http_client.py
import requests
from typing import Dict, Any


class HttpClient:
    def __init__(self, base_url: str):
        self.base_url = base_url

    def post(self, url: str, body: Dict[str, Any], token: str) -> Dict[str, Any]:
        headers = {
            'Authorization': f'Bearer {token}',
            'Content-Type': 'application/json',
        }
        
        response = requests.post(url, json=body, headers=headers)
        response.raise_for_status()
        
        return response.json()

    def get(self, url: str, token: str) -> Dict[str, Any]:
        headers = {
            'Authorization': f'Bearer {token}',
        }
        
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        
        return response.json()

    def get_base_url(self) -> str:
        return self.base_url
